CONTRIBUTING.md

```diff
@@ -1,19 +1,61 @@
+Contributing to Snap is a snap (pun intended :turtle:).
+
+Snap is Apache 2.0 licensed and accepts contributions via GitHub. This document will cover how to contribute to the project and report issues. Testing is covered in [BUILD_AND_TEST.md](docs/BUILD_AND_TEST.md).
+
+## Topics
+
+* [Reporting Security Issues](#reporting-security-issues)
+* [Reporting Issues or Feature Requests](#reporting-issues-or-feature-requests)
+* [Contributing Code](#contributing-code)
+* [Notes on GitHub Usage](#notes-on-github-usage)
+
+## Reporting Security Issues
+
+The Snap team take security very seriously. If you have any issue regarding security,
+please notify us by sending an email to snap-security@intel.com. **Do not create a GitHub issue.** We will follow up with you promptly with more information and a plan for remediation. While we are not offering a security bounty, we would love to send some Snap swag your way along with our deepest gratitude for your assistance in making Snap a more secure product.
+
+## Reporting Issues or Feature Requests
+
+Your contribution through issues is very beneficial to the project. We appreciate all reported issues or new requests. They come in a few different forms described below. Before opening a new issue, we appreciate you reviewing open issues to see if there are any similar requests. If there is a match, comment with a +1, "Also seeing this issue" or something like that. If any environment details differ, please add those with your comment to the matching issue.
+
+When **reporting an issue**, details are key. Include the following:
+- OS version
+- Snap version
+- Environment details (virtual, physical, etc.)
+- Steps to reproduce
+- Expected results
+- Actual results
+
+When **requesting a feature**, context is key. Some of the questions we want to answer are:
+- What does this allow a user to do that they cannot do now?
+- What environment is this meant for (containerized, virtualized, bare metal)?
+- How urgent is the need (nice to have feature or need to have)?
+- Does this align with the goals of Snap?
+
+When **proposing a new approach**, understanding the use case is key. We organize these requests into a RFC process. RFCs must include:
+- A start and end date for comments
+- A set of required participants
+- A proposal that includes what is in and out of scope
+
+### What is an RFC?
+
+The Snap maintainers use RFCs, loosely based off the [EITF practice](https://www.ietf.org/rfc.html), to discuss proposed architectural and organizational improvements to the project. While most RFCs will originate from the maintainers, we welcome all users suggesting improvements through the RFC process.
+
+When **proposing a new approach to solving a challenge**, whether technical or organizational, open a request for comment (RFC). If you are ever unsure whether you should do so, open a regular issue and maintainers will help you groom it into a RFC.
```

Comment:
Do people who are not maintainers have the ability to open an issue other than a `regular issue`? What is the difference between a regular issue and a RFC issue -- isn't it just that an RFC has the ``type/rfc`` tag. I thought only maintainers could add tags to issues.